judge/utils/tests/test_file_cache.py

```diff
@@ -0,0 +1,111 @@
+import gzip
+import os
+import shutil
+import tempfile
+from io import UnsupportedOperation
+from urllib.parse import urljoin
+
+from django.test import SimpleTestCase
+
+from judge.utils.file_cache import HashFileCache
+
+
+class FileCacheTestCase(SimpleTestCase):
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.cache = HashFileCache(self.temp_dir, 'https://example.com')
+        self.gzip_temp_dir = tempfile.mkdtemp()
+        self.gzip_cache = HashFileCache(self.gzip_temp_dir, 'https://example.com', gzip=True)
+
+        # This acts as a test for create() as well as setup for other tests
+        self.cache.create('foo')
+        self.cache.create('foo')
```

Comment:
Why do this twice?

Reply:
To make sure that creating a file that already exists doesn't throw the error. It tests the `False` outcome of [this](https://github.com/DMOJ/online-judge/blob/master/judge/utils/file_cache.py#L17).

Reply:
Shouldn't this be a separate test case?

Reply:
Agreed.